test/lib/types/library/libraryType.js

```diff
@@ -0,0 +1,108 @@
+const test = require("ava");
+const sinon = require("sinon");
+const mock = require("mock-require");
+
+test.beforeEach((t) => {
+	t.context.LibraryFormatterFormatStub = sinon.stub().returns(Promise.resolve());
+	t.context.LibraryFormatterStub = sinon.stub().returns({
+		format: t.context.LibraryFormatterFormatStub
+	});
+	mock("../../../../lib/types/library/LibraryFormatter", t.context.LibraryFormatterStub);
+
+	t.context.LibraryBuilderBuildStub = sinon.stub().returns(Promise.resolve());
+	t.context.LibraryBuilderStub = sinon.stub().returns({
+		build: t.context.LibraryBuilderBuildStub
+	});
+	mock("../../../../lib/types/library/LibraryBuilder", t.context.LibraryBuilderStub);
+
+	t.context.libraryType = mock.reRequire("../../../../lib/types/library/libraryType");
+});
+
+test.afterEach.always((t) => {
+	mock.stopAll();
+	sinon.restore();
+});
+
+test.serial("libraryType.format", (t) => {
+	const {
+		libraryType,
+		LibraryFormatterStub, LibraryFormatterFormatStub
+	} = t.context;
+
+	const project = {"fake": "project"};
+
+	const returnValue = libraryType.format(project);
+
+	t.is(LibraryFormatterStub.callCount, 1,
+		"LibraryFormatter should be called once");
```

Comment:
I'm wondering about the 'should' here and in other places - does it mean the LibraryFormatter is called or not?

Reply:
This is a test file (the path starts with `test/`). This assertion *expects* the LibraryFormatter to be called. If it is not, the assertion fails.